### PR TITLE
Add a "generic" perf-test for supernodal sptrsv

### PR DESCRIPTION
**Pull request by iyamazaki**
- We added a more "generic" perf-test for supernodal sptrsv 
   (does not depend on SuperLU/Cholmod)
- We cleaned/restructured the setup to better support the new interface

### CODE REVIEW

**Comment by iyamazaki** [comment]
The output from spot-check on White:

#######################################################
PASSED TESTS
#######################################################
cuda-10.0.130-Cuda_Serial-release build_time=1129 run_time=859
cuda-9.2.88-Cuda_OpenMP-release build_time=1334 run_time=672
gcc-6.4.0-OpenMP_Serial-release build_time=586 run_time=1081
gcc-7.2.0-OpenMP-release build_time=384 run_time=561
gcc-7.2.0-OpenMP_Serial-release build_time=594 run_time=1258
gcc-7.2.0-Serial-release build_time=279 run_time=468
gcc-7.4.0-OpenMP-release build_time=401 run_time=523
ibm-16.1.0-Serial-release build_time=1552 run_time=774
#######################################################
FAILED TESTS
#######################################################

**Comment by ndellingwood** [comment]
@iyamazaki I'm rerunning spot-check with this PR merged into develop, will get this merged after tests finish.

**Comment by ndellingwood** [comment]
Spot-check:

```
Running on machine: kokkos-dev-2
KokkosKernels Repository Status:  add3a569bef85f797b69bdbdd0b3fbfa9fa853ae Merge branch 'develop' into pr-619

Kokkos Repository Status:  c0a0333d3e419fb3e79ea9d1e7ee187e1fed28ad Merge pull request #2901 from msimberg/hpx-backend-fixes


Going to test compilers:  gcc/7.3.0 gcc/8.3.0 gcc/9.1 intel/18.0.5 clang/8.0 cuda/10.1 cuda/9.2
Testing compiler gcc/7.3.0
  Starting job gcc-7.3.0-OpenMP-release
kokkos options: 
kokkos devices: OpenMP
kokkos cxx: -O3 -Wall -Wshadow -pedantic -Werror -Wsign-compare -Wtype-limits -Wignored-qualifiers -Wempty-body -Wclobbered -Wuninitialized 
  PASSED gcc-7.3.0-OpenMP-release
Testing compiler gcc/8.3.0
  Starting job gcc-7.3.0-Pthread-release
kokkos options: 
kokkos devices: Pthread
kokkos cxx: -O3 -Wall -Wshadow -pedantic -Werror -Wsign-compare -Wtype-limits -Wignored-qualifiers -Wempty-body -Wclobbered -Wuninitialized 
  PASSED gcc-7.3.0-Pthread-release
Testing compiler gcc/9.1
  Starting job gcc-8.3.0-Serial-release
kokkos options: 
kokkos devices: Serial
kokkos cxx: -O3 -Wall -Wshadow -pedantic -Werror -Wsign-compare -Wtype-limits -Wignored-qualifiers -Wempty-body -Wclobbered -Wuninitialized 
  PASSED gcc-8.3.0-Serial-release
  Starting job gcc-9.1-OpenMP-release
kokkos options: 
kokkos devices: OpenMP
kokkos cxx: -O3 -Wall -Wshadow -pedantic -Werror -Wsign-compare -Wtype-limits -Wignored-qualifiers -Wempty-body -Wclobbered -Wuninitialized 
  PASSED gcc-9.1-OpenMP-release
Testing compiler intel/18.0.5
  Starting job gcc-9.1-Serial-release
kokkos options: 
kokkos devices: Serial
kokkos cxx: -O3 -Wall -Wshadow -pedantic -Werror -Wsign-compare -Wtype-limits -Wignored-qualifiers -Wempty-body -Wclobbered -Wuninitialized 
  PASSED gcc-9.1-Serial-release
Testing compiler clang/8.0
  Starting job intel-18.0.5-OpenMP-release
kokkos options: 
kokkos devices: OpenMP
kokkos cxx: -O3 -Wall -Wshadow -pedantic -Werror -Wsign-compare -Wtype-limits -Wuninitialized 
  PASSED intel-18.0.5-OpenMP-release
  Starting job clang-8.0-Cuda_OpenMP-release
kokkos options: 
kokkos devices: Cuda,OpenMP
kokkos cxx: -O3 -Wall -Wshadow -pedantic -Werror -Wsign-compare -Wtype-limits -Wuninitialized -Wno-pass-failed 
  PASSED clang-8.0-Cuda_OpenMP-release
Testing compiler cuda/10.1
  Starting job clang-8.0-Pthread_Serial-release
kokkos options: 
kokkos devices: Pthread,Serial
kokkos cxx: -O3 -Wall -Wshadow -pedantic -Werror -Wsign-compare -Wtype-limits -Wuninitialized -Wno-pass-failed 
  PASSED clang-8.0-Pthread_Serial-release
Testing compiler cuda/9.2
  Starting job cuda-10.1-Cuda_OpenMP-release
kokkos options: 
kokkos devices: Cuda,OpenMP
kokkos cxx: -O3 -Wall -Wshadow -pedantic -Werror -Wsign-compare -Wtype-limits -Wuninitialized 
  PASSED cuda-10.1-Cuda_OpenMP-release
  Starting job cuda-9.2-Cuda_Serial-release
kokkos options: 
kokkos devices: Cuda,Serial
kokkos cxx: -O3 -Wall -Wshadow -pedantic -Werror -Wsign-compare -Wtype-limits -Wuninitialized 
  PASSED cuda-9.2-Cuda_Serial-release
#######################################################
PASSED TESTS
#######################################################
clang-8.0-Cuda_OpenMP-release build_time=136 run_time=155
clang-8.0-Pthread_Serial-release build_time=49 run_time=175
cuda-10.1-Cuda_OpenMP-release build_time=148 run_time=221
cuda-9.2-Cuda_Serial-release build_time=145 run_time=294
gcc-7.3.0-OpenMP-release build_time=40 run_time=285
gcc-7.3.0-Pthread-release build_time=40 run_time=80
gcc-8.3.0-Serial-release build_time=37 run_time=71
gcc-9.1-OpenMP-release build_time=46 run_time=53
gcc-9.1-Serial-release build_time=41 run_time=73
intel-18.0.5-OpenMP-release build_time=136 run_time=114
```

File: perf_test/sparse/KokkosSparse_sptrsv_supernode.cpp [hunk 1]
```diff
@@ -0,0 +1,445 @@
+/*
+//@HEADER
+// ************************************************************************
+//
+//               KokkosKernels 0.9: Linear Algebra and Graph Kernels
```
Comment (ndellingwood) [inline perf_test/sparse/KokkosSparse_sptrsv_supernode.cpp]: @iyamazaki The header info will need to be updated with the new 3.0 copyright info, you can look at another file and just copy+replace the header here.

**Pull request merged**